shared/gh/js/views/tenant-admin/gh.admin-batch-edit.js

```diff
@@ -203,13 +206,25 @@ define(['gh.core', 'gh.constants', 'moment', 'moment-timezone', 'gh.calendar', '
         });
         // Get the number of weeks in the term
         var weeksInTerm = gh.utils.getWeeksInTerm(term);
-        // Add a new event row for every week in the term
-        for (weeksInTerm; weeksInTerm !== 0; weeksInTerm--) {
-            // Add new event rows
+
+        // The container where the event rows should be added to
+        var $eventContainer = $(this).closest('.gh-batch-edit-events-container').find('tbody');
+
+        // This recursive loop is necessary because `addNewEventRow` is asynchronous. On its own, this
+        // wouldn't be a huge problem, but `addNewEventRow` checks the DOM to see how many rows have
+        // been added previously. If this were executed synchronously 8 times, it would add 8 events
+        // all of them starting on the same day in week 1
+        var addRow = function(i) {
             addNewEventRow(ev, {
```

Comment:
It would be nicer if we could just pass the number of desired rows into `addNewEventRow` without having to worry about the asynchronous nature and what's in the DOM already, and without having to do things like initialize jEditable for every row individually.